metrics/logpoller.go

```diff
@@ -0,0 +1,54 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type QueryType string
+
+const (
+	Create QueryType = "create"
+	Read   QueryType = "read"
+	Del    QueryType = "delete"
+)
+
+var (
+	sqlLatencyBuckets = []float64{
+		float64(1 * time.Millisecond),
+		float64(5 * time.Millisecond),
+		float64(10 * time.Millisecond),
+		float64(20 * time.Millisecond),
+		float64(30 * time.Millisecond),
+		float64(40 * time.Millisecond),
+		float64(50 * time.Millisecond),
+		float64(60 * time.Millisecond),
+		float64(70 * time.Millisecond),
+		float64(80 * time.Millisecond),
+		float64(90 * time.Millisecond),
+		float64(100 * time.Millisecond),
+		float64(200 * time.Millisecond),
+		float64(300 * time.Millisecond),
+		float64(400 * time.Millisecond),
+		float64(500 * time.Millisecond),
+		float64(750 * time.Millisecond),
+		float64(1 * time.Second),
+		float64(2 * time.Second),
+		float64(5 * time.Second),
+	}
+	LpQueryDuration = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "log_poller_query_duration",
+		Help:    "Measures duration of Log Poller's queries fetching logs",
+		Buckets: sqlLatencyBuckets,
+	}, []string{"chainFamily", "chainID", "query", "type"})
+	LpQueryDataSets = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "log_poller_query_dataset_size",
+		Help: "Measures size of the datasets returned by Log Poller's queries",
+	}, []string{"chainFamily", "chainID", "query", "type"})
+	LpLogsInserted = promauto.NewCounterVec(prometheus.CounterOpts{
```

Comment:
I guess you can have lpBlockInserted as well https://github.com/smartcontractkit/chainlink-integrations/blob/9f4f882c2e61a6bf5353148bab5f887f3094a25e/evm/logpoller/observability.go#L63

Metric won't be populated for chains that LogPoller doesn't insert any blocks to database (e.g. Solana)